### PR TITLE
Fixes #17482: Webapp block at start on inventory processing

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17482

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala [hunk 1]
```diff
@@ -109,19 +109,27 @@ object Watchers {
...
-        override def start()(implicit executionContext: ExecutionContext) = {
-          this.watch(root, 0)
-          executionContext.execute(new Runnable {
-            override def run() = {
-              try {
-                blocking { Iterator.continually(service.take()).foreach(process) }
-              } catch {
+        // Execution context is not used here - binding to scala default global
+        @silent("private default argument.*is never used")
+        override def start()(implicit executionContext: ExecutionContext = scala.concurrent.ExecutionContext.global): Unit = {
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala:114]: Have you tried removing the execution context ?

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala (hunk 1 above)
Comment (fanf) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala]: it's in parent method signature, I can't. But I moved it to call site, it avoids half of them.

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala]: Ha yes, we override better-files file monitor

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**